Python/Product/VSInterpreters/Interpreter/CondaEnvironmentFactoryProvider.cs

```diff
@@ -285,25 +285,17 @@ class CondaInfoResult {
             public string[] EnvironmentRootFolders = null;
         }
 
-        private List<PythonInterpreterInformation> FindCondaEnvironments(string condaPath) {
-            var found = new List<PythonInterpreterInformation>();
-            var watchFolders = new HashSet<string>();
-
+        private IEnumerable<PythonInterpreterInformation> FindCondaEnvironments(string condaPath) {
             var condaInfoResult = ExecuteCondaInfo(condaPath);
             if (condaInfoResult != null) {
-                foreach (var folder in condaInfoResult.EnvironmentFolders) {
-                    if (!Directory.Exists(folder)) {
-                        continue;
-                    }
-
-                    PythonInterpreterInformation env = CreateEnvironmentInfo(folder);
-                    if (env != null) {
-                        found.Add(env);
-                    }
-                }
+                return condaInfoResult.EnvironmentFolders
+                    .AsParallel()
+                    .Where(folder => Directory.Exists(folder))
+                    .Select(folder => CreateEnvironmentInfo(folder))
+                    .Where(env => env != null);
```

Comment:
Since this method uses file system, it is better to call `ToList` at the end to avoid multiple enumerations